src/main/java/reposense/git/GitClone.java

```diff
@@ -119,4 +169,32 @@ public static void cloneFromBareAndUpdateBranch(Path rootPath, RepoConfiguration
     private static String getCloneBareCommand(RepoConfiguration config, String outputFolderName) {
         return "git clone --bare " + config.getLocation() + " " + outputFolderName;
     }
+
+    /**
+     * Constructs the command to clone a bare repo specified in the {@code config}
+     * into the folder {@code outputFolderName}.
+     */
+    private static String getCloneShallowBareCommand(RepoConfiguration config,
+                                                    String outputFolderName, Date shallowSinceDate) {
+        return "git clone --bare --shallow-since=\"" + shallowSinceDate + "\" "
+                + config.getLocation() + " " + outputFolderName;
+    }
+
+    /**
+     * Constructs the command to clone a bare repo specified in the {@code config}
```

Comment:
You have two functions with a duplicate comment header. Better to modify to show what each function exactly does. Will be easier for other devs to understand.